Add `ignore_spaces` option to `ini_file` to ignore spacing changes

jikamens opened the pull request:
##### SUMMARY

Add a new `ignore_spaces` option to the `ini_file` module which, if true, prevents the module from changing a line in a file if the only thing that would change by doing so is whitespace before or after the `=`.

Also add  unit and integration test cases for this new functionality.

Also fixes two bugs in the recently added tests for symlinked ini files: they weren't being executed at all, and they were creating their temporary files in the wrong place.

Fixes #7202

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ini_file

felixfontein commented on plugins/modules/ini_file.py:
```diff
@@ -98,6 +99,11 @@
     default: false
+  ignore_spaces:
+    description:
+      - Don't change a line if doing so would only add or remove spaces before or after the '=' symbol.
```
```suggestion
      - Do not change a line if doing so would only add or remove spaces before or after the V(=) symbol.
```

felixfontein commented on plugins/modules/ini_file.py:
```diff
@@ -98,6 +99,11 @@
+    description:
+      - Don't change a line if doing so would only add or remove spaces before or after the '=' symbol.
+    type: bool
+    default: false
```
```suggestion
    default: false
    version_added: 7.5.0
```

felixfontein reviewed (commented):
Thanks for your contribution! Some first quick comments.

felixfontein commented:
> There were previously no tests for `ini_file` at all

That's not true, ini_file has quite a few integration tests. See https://github.com/ansible-collections/community.general/tree/main/tests/integration/targets/ini_file/.

jpmens commented:
shipit

jikamens commented:
> > There were previously no tests for `ini_file` at all
> 
> That's not true, ini_file has quite a few integration tests. See https://github.com/ansible-collections/community.general/tree/main/tests/integration/targets/ini_file/.

Oh. I'm sorry for the error! In that case don't ship this until I have a chance to look at the integration tests and augment them appropriately.

jikamens commented:
I fixed several independent issues with the `ini_file` integration tests, incorporated the two changes requested above, and added integration tests for `ignore_spaces`.

felixfontein commented:
@jikamens thanks a lot for implementing this, and for improving/fixing the tests!
@jpmens thanks a lot for reviewing!
